bluesky/broker_callbacks.py

```diff
@@ -87,3 +90,47 @@ def f(name, stop_doc):
             callback('event', e)
         callback('stop', stop_doc)
     return f
+
+
+def exporter_factory(field, template):
+    """
+    Build a function that, given a header, exports tiff files.
+
+    The file names will incorporate the contents of the Header.
+
+    Parameters
+    ----------
+    field : str
+        a data key, e.g., 'pe1_image_lightfield'
+    template : str
+        A templated file path, where curly brackets will be filled in with
+        the attributes of 'h', a Header, and 'N', a sequential number.
+        e.g., "dir/scan{h.start.scan_id}_by_{h.start.experimenter}_{N}.tiff"
+
+    Returns
+    -------
+    f : function
+        a function that accepts a header and saves TIFF files
+    """
+    # validate user input
+    if not '{N}' in template:
+        raise ValueError("template must include '{N}'")
+
+    def f(h):
+        imgs = get_images(h, field)
+        # Fill in h, defer filling in N.
+        _template = template.format(h=h, N='{N}')
+        for i, img in enumerate(imgs):
```

Comment:
This should generate and check if any of the files exist.  If _any_ exist it should raise.